cibuildwheel/ios.py

```diff
@@ -0,0 +1,635 @@
+    # well. Rather than duplicate the location of the URL of macOS installers,
+    # load the macos configurations, determine the macOS configuration that
+    # matches the platform we're building, and embed that URL in the parsed iOS
+    # configuration.
+    macos_python_configs = resources.read_python_configs("macos")
+
+    def build_url(item: dict[str, str]) -> str:
+        # The iOS item will be something like cp313-ios_arm64_iphoneos. Drop
+        # the iphoneos suffix, then replace ios with macosx to yield
+        # cp313-macosx_arm64, which will be a macOS configuration item.
+        macos_identifier = item["identifier"].rsplit("_", 1)[0]
+        macos_identifier = macos_identifier.replace("ios", "macosx")
+        matching = [
+            config for config in macos_python_configs if config["identifier"] == macos_identifier
+        ]
+        return matching[0]["url"]
+
+    # Load the platform configuration
+    if sdk:
+        full_python_configs = resources.read_python_configs(sdk)
+    else:
+        full_python_configs = resources.read_python_configs("iphoneos")
+        full_python_configs += resources.read_python_configs("iphonesimulator")
+
+    # Build the configurations, annotating with macOS URL details.
+    python_configurations = [
+        PythonConfiguration(
+            **item,
+            build_url=build_url(item),
+        )
+        for item in full_python_configs
+    ]
+
+    # Filter out configs that don't match any of the selected architectures
+    python_configurations = [
+        c
+        for c in python_configurations
+        if any(c.identifier.rsplit("_", 1)[0].endswith(a.value) for a in architectures)
+    ]
+
+    # Skip builds as required by BUILD/SKIP
+    python_configurations = [c for c in python_configurations if build_selector(c.identifier)]
+
+    return python_configurations
+
+
+def install_host_cpython(tmp: Path, config: PythonConfiguration, free_threading: bool) -> Path:
+    if free_threading:
+        msg = "Free threading builds aren't available for iOS (yet)"
+        raise ValueError(msg)
+
+    # Install an iOS build of CPython
+    ios_python_tar_gz = config.url.rsplit("/", 1)[-1]
+    extension = ".tar.gz"
+    assert ios_python_tar_gz.endswith(extension)
+    installation_path = CIBW_CACHE_PATH / ios_python_tar_gz[: -len(extension)]
+    with FileLock(str(installation_path) + ".lock"):
+        if not installation_path.exists():
+            downloaded_tar_gz = tmp / ios_python_tar_gz
+            download(config.url, downloaded_tar_gz)
+            installation_path.mkdir(parents=True, exist_ok=True)
+            call("tar", "-C", installation_path, "-xf", downloaded_tar_gz)
+            downloaded_tar_gz.unlink()
+
+    return installation_path
+
+
+def cross_virtualenv(
+    py_version: str,
+    host_python: Path,
+    multiarch: str,
+    build_python: Path,
+    venv_path: Path,
+    dependency_constraint_flags: Sequence[PathOrStr],
+) -> dict[str, str]:
```

Comment:
Not sure what I can do here... these are all required arguments, and factoring out a data structure for exactly one invocation seems excessive.